drivers/linstorvhdutil.py

```diff
@@ -120,14 +110,45 @@ def wrapper(*args, **kwargs):
             remote_args = {str(key): str(value) for key, value in remote_args.items()}
 
             try:
-                def remote_call():
-                    host_ref = self._get_readonly_host(vdi_uuid, device_path, node_names)
-                    return call_remote_method(self._session, host_ref, remote_method, device_path, remote_args)
-                response = util.retry(remote_call, 5, 2)
-            except Exception as remote_e:
-                self._raise_openers_exception(device_path, local_e or remote_e)
+                host_ref_attached = list(util.get_hosts_attached_on(self._session, [vdi_uuid]))[0]
```

Comment:
The call to `util.get_hosts_attached_on()` only "works" for a leaf VDI. Base copies don't have this information stored on them. It's out of scope for this PR but eventually we might want to check for leaves of our VDI.